crates/red_knot_test/src/parser.rs

```diff
@@ -328,6 +334,13 @@ impl<'s> Parser<'s> {
         // We never pop the implicit root section.
         let section = self.stack.top();
 
+        if captures.name("end").unwrap().is_empty() {
+            return Err(anyhow::anyhow!(
+                "Unterminated code block at index {}.",
```

Comment:
This will always say "at index 0" no matter where in the file the unterminated code block starts, because of the way we use `Cursor` to advance through the source text; we are always matching this regex at the start of a string. (Checked this experimentally.)

I _think_ (not tested) that `self.cursor.token_len()` will give you the offset into the original source text that we've consumed so far, and still let you easily provide this info in the error. If for some reason that doesn't work, I'm also OK with just not providing this data. In practice, I'm not sure how useful it is, because I _think_ if any code block other than the last one is unterminated, you'll just get a situation where the contents between the unterminated code block and the next will get swallowed, and it'll end up reporting a later code block as unterminated, not the one you actually need to fix?